src/App.tsx

```diff
@@ -1,60 +1,129 @@
 import './App.scss';
+import { FormEvent, useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [userError, setUserError] = useState(false);
+  const maxId = Math.max(...todos.map(todo => todo.id));
+
+  const addTodo = () => {
+    const newTodo = {
+      id: maxId + 1,
+      title,
+      userId,
+      completed: false,
+      user: getUser(userId),
+    };
+
+    if (newTodo && title && userId) {
+      return todos.push(newTodo);
+    }
+
+    return todos;
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      setTitleError(true);
+    }
+
+    if (!userId) {
+      setUserError(true);
+    }
+
+    if (!title || !userId) {
+      return;
+    }
+
+    setTitle('');
+    setUserId(0);
+    setTitleError(false);
+    setUserError(false);
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form action="/api/users" method="POST" onSubmit={handleSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="title">
+            {'Title: '}
+            <input
+              type="text"
+              name="title"
+              id="title"
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              value={title}
+              onChange={(event) => {
+                setTitle(event.target.value.replace(
+                  /[^\w\s\p{Script=Cyrillic}]/gui, '',
+                ));
```

Comment:
Cool 👍 but move it to handler for example handleTitleChange and also move regex to a variable